src/Microsoft.Azure.Functions.Extensions.Mcp/Trigger/McpToolListener.cs

```diff
@@ -64,7 +69,46 @@ public async Task<CallToolResult> RunAsync(CallToolRequestParams callToolRequest
                 }
             ]
         };
+    }
+
+    internal static void ValidateArgumentsHaveRequiredProperties(ICollection<IMcpToolProperty> properties, CallToolRequestParams callToolRequest)
+    {
+        var requiredProps = properties?
+                            .Where(p => p.Required)
+                            .ToArray();
+
+        if (requiredProps is null || requiredProps.Length == 0)
+        {
+            return;
+        }
+
+        var missing = new List<string>();
+        var args = callToolRequest?.Arguments;
 
+        foreach (var prop in requiredProps)
+        {
+            if (args == null
+                || !args.TryGetValue(prop.PropertyName, out JsonElement value)
+                || IsValueNullOrUndefined(value))
+            {
+                missing.Add(prop.PropertyName);
+            }
+        }
+        if (missing.Count > 0)
```

Comment:
nit: new line between branches
```suggestion

        if (missing.Count > 0)
```